src/analysis.jl

```diff
@@ -25,47 +25,45 @@ function code(c::DynamicCall; optimize = false)
   codeinfo = code_typed(c.f, argtypes(c), optimize = optimize)
   @assert length(codeinfo) == 1
   codeinfo = codeinfo[1]
-  linearize!(codeinfo[1])
+  # linearize!(codeinfo[1])
   return codeinfo
 end
 
-struct StaticCall <: Call
-  method_instance::MethodInstance
-end
-
-argtypes(c::StaticCall) = Tuple{c.method_instance.specTypes.parameters[2:end]...}
-types(c::StaticCall) = c.method_instance.specTypes
-method(c::StaticCall) = c.method_instance.def
-
-method_expr(c::StaticCall) = method_expr(method(c).name, argtypes(c))
-
-function code(c::StaticCall; optimize = false)
-  # TODO static call graph can only be computed with optimize=true, so analyzing with optimized=false will skip inlined methods
-  codeinfo = get_code_info(c.method_instance, optimize=true)
-  linearize!(codeinfo[1])
-  return codeinfo
-end
+# struct StaticCall <: Call
+#   method_instance::MethodInstance
+# end
+#
+# argtypes(c::StaticCall) = Tuple{c.method_instance.specTypes.parameters[2:end]...}
+# types(c::StaticCall) = c.method_instance.specTypes
+# method(c::StaticCall) = c.method_instance.def
+#
+# method_expr(c::StaticCall) = method_expr(method(c).name, argtypes(c))
+#
+# function code(c::StaticCall; optimize = false)
+#   # TODO static call graph can only be computed with optimize=true, so analyzing with optimized=false will skip inlined methods
+#   codeinfo = get_code_info(c.method_instance, optimize=true)
+#   linearize!(codeinfo[1])
+#   return codeinfo
+# end
 
 function eachline(f, code, line = -1)
-  for l in code.code
-    if l isa LineNumberNode
-      line = l.line
-    else
-      f(line, l)
-    end
+  for (i, l) in enumerate(code.code)
+    line = code.linetable[code.codelocs[i]].line
+    f(line, l)
   end
 end
 
 exprtype(code, x) = typeof(x)
-exprtype(code, x::Expr) = x.typ
+exprtype(code, x::Expr) = Union{}
```

Comment:
I don't think this should be called any more, so perhaps we can just remove it. The same information should be in the SSAValue type.

Reply:
True, I meant to remove that. There's a bit of cruft around anyways.